src/reader.rs

```diff
@@ -91,3 +50,49 @@ impl ChunkReader for File {
         Ok(BufReader::new(self.try_clone()?))
     }
 }
+
+#[allow(clippy::len_without_is_empty)]
+pub trait AsyncChunkReader: Send {
```

Comment:
Async complement to ChunkReader

Ideally these two interfaces will be all we need to interact with the file sources